Applications/ApplicationsLib/UncoupledProcessesTimeLoop.h

```diff
@@ -0,0 +1,365 @@
+    };
+
+    /*! Creates a new instance of PerProcessData from the given data.
+     *
+     * \tparam ODETag tag indicating the type of ODE to be solved.
+     *
+     * \param nl_solver the nonlinear solver to be used
+     * \param ode_sys   the ODE (i.e. the process) to be integrated in time
+     * \param time_disc the time discretization used for integrating \c ode_sys
+     *
+     * \note Currently the \c ODETag is automatically inferred from the given
+     *       \c ody_sys. This works as long as \c Process derives from
+     *       \c ODESystem<Matrix, Vector, ODETag>, i.e. as long we only deal with
+     *       one type of ODE. When we introduce more types, this method will have
+     *       to be extended slightly.
+     */
+    template<NumLib::ODESystemTag ODETag>
+    static
+    PerProcessData
+    makePerProcessData(
+            NumLib::NonlinearSolverBase<Matrix, Vector>& nl_solver,
+            NumLib::ODESystem<Matrix, Vector, ODETag,
+                NumLib::NonlinearSolverTag::Picard>& ode_sys,
+            TimeDisc& time_disc)
+    {
+        using Tag = NumLib::NonlinearSolverTag;
+        // A concrete Picard solver
+        using NLSlvPicard = NumLib::NonlinearSolver<Matrix, Vector, Tag::Picard>;
+        // A concrete Newton solver
+        using NLSlvNewton = NumLib::NonlinearSolver<Matrix, Vector, Tag::Newton>;
+
+        if (auto* nlslv_picard = dynamic_cast<NLSlvPicard*>(&nl_solver))
+        {
+            // The Picard solver can also work with a Newton-ready ODE,
+            // because the Newton ODESystem derives from the Picard ODESystem.
+            // So no further checks are needed here.
+
+            return PerProcessData{ *nlslv_picard, time_disc, ode_sys };
+        }
+        else if (auto* nlslv_newton = dynamic_cast<NLSlvNewton*>(&nl_solver))
+        {
+            // The Newton-Raphson method needs a Newton-ready ODE.
+
+            using ODENewton = NumLib::ODESystem<Matrix, Vector, ODETag, Tag::Newton>;
+            if (auto* ode_newton = dynamic_cast<ODENewton*>(&ode_sys))
+            {
+                return PerProcessData{ *nlslv_newton, time_disc, *ode_newton };
+            }
+            else {
+                ERR("You are trying to solve a non-Newton-ready ODE with the"
+                    " Newton-Raphson method. Aborting");
+                std::abort();
+            }
+        }
+        else {
+            ERR("Encountered unknown nonlinear solver type. Aborting");
+            std::abort();
+        }
+    }
+
+    //! Constructs and returns a \c vector of PerProcessData from the given
+    //! ProjectData.
+    std::vector<PerProcessData> initInternalData(ProjectData& project);
+
+    //! Sets and returns initial conditions for the given ProjectData and
+    //! PerProcessData.
+    std::vector<Vector> setInitialConditions(
+            ProjectData& project, double const t0,
+            std::vector<PerProcessData>& per_process_datas);
```

Comment:
C'mon! What is datas? vector of childs?
per_process_data is something which is holding data per process already.
Rename PerProcessData -> ProcessData, keep per_process_data.